pulser/waveforms.py

```diff
@@ -33,6 +34,17 @@
 from pulser.parametrized.decorators import parametrize
 from pulser.json.utils import obj_to_dict
 
+if version_info[:2] == (3, 7):  # pragma: no cover
+    try:
+        from backports.cached_property import cached_property
+    except ImportError:
+        raise ImportError(
+            "Using pulser with Python version 3.7 requires the"
+            " `backports.cached-property` module. Install it by running"
+            " `pip install backports.cached-property`.")
+else:  # pragma: no cover
+    from functools import cached_property  # type: ignore
```

Comment:
`mypy` CI tests are made with python 3.8, so you don't need the `# type: ignore`, but rather need it on line 39.

Reply:
> `mypy` CI tests are made with python 3.8, so you don't need the `# type: ignore`, but rather need it on line 39.

```
pulser\waveforms.py:39: error: unused 'type: ignore' comment
pulser\waveforms.py:46: error: Incompatible import of "cached_property" (imported name has type "Type[functools.cached_property[Any]]", local name has type "Type[backports.cached_property.cached_property[Any]]")
```

The `# type: ignore` seems correctly placed to me. It is exactly what I did with `sequence.py`, for the same reasons.

I think the `Library stubs not installed for "backports.cached_property" (or incompatible with Python 3.8)` error comes from elsewhere.

There is a reason why this works with `sequence.py` and not `waveforms.py` but I can't figure it out.

Reply:
The reason why we put the `# type: ignore` on the second import is because it complains if you import an object with the same name twice. Since we are testing with mypy on Python 3.8, you want `from functools import chached_property` to come first, and then have the `# type: ignore` on the `from backports.cached_property ...`. Can you try this please?